src/renderer/components/ReactFlowBox.tsx

```diff
@@ -310,6 +327,220 @@ const ReactFlowBox = memo(({ wrapperRef, nodeTypes, edgeTypes }: ReactFlowBoxPro
+                    _hover={{ color: useColorModeValue('black', 'white') }}
+                    style={{
+                        color: useColorModeValue('gray.500', 'gray.300'),
+                        cursor: 'pointer',
+                        display: searchQuery ? undefined : 'none',
+                        fontSize: '66%',
+                    }}
+                    onClick={() => setSearchQuery('')}
+                >
+                    <CloseIcon />
+                </InputRightElement>
+            </InputGroup>
+            <Box
+                h="auto"
+                maxH={400}
+                overflowY="scroll"
+                p={1}
+            >
+                {[...byCategories].map(([category, categoryNodes]) => {
+                    const accentColor = getNodeAccentColors(category);
+                    return (
+                        <Box key={category}>
+                            <HStack
+                                borderRadius="md"
+                                mx={1}
+                                py={0.5}
+                            >
+                                <IconFactory
+                                    accentColor={accentColor}
+                                    boxSize={3}
+                                    icon={category}
+                                />
+                                <Text fontSize="xs">{category}</Text>
+                            </HStack>
+                            {[...categoryNodes].map((node) => (
+                                <HStack
+                                    _hover={{ backgroundColor: 'gray.700' }}
+                                    borderRadius="md"
+                                    key={node.schemaId}
+                                    mx={1}
+                                    px={2}
+                                    py={0.5}
+                                    onClick={() => {
+                                        const reactFlowBounds =
+                                            wrapperRef.current!.getBoundingClientRect();
+                                        const { x, y } = menuRef.current!.getBoundingClientRect();
+                                        const position = reactFlowInstance.project({
+                                            x: x - reactFlowBounds.left,
+                                            y: y - reactFlowBounds.top,
+                                        });
+                                        const nodeToMake: NodeProto = {
+                                            position,
+                                            data: {
+                                                schemaId: node.schemaId,
+                                            },
+                                            nodeType: node.nodeType,
+                                        };
+                                        createNode(nodeToMake, (newNodes) => {
```

Comment:
The `onNodeCreate` function might be a problem because it is called from within the supplier function of a `setState` function (`setNodes` to be specific). It feels like doing that is an anti-pattern if not incorrect. However, the React docs don't seem to mention this.